File: benchmarks/go/14_recursive_list_sum.go
// Recursive List Sum Benchmark - Go Implementation
// Recursive sum of a slice

package main

import "fmt"

func sumSlice(slice []int, idx int) int {
	if idx >= len(slice) {
		return 0
	}
	return slice[idx] + sumSlice(slice, idx+1)
}

func main() {
	slice := make([]int, 1000)
	for i := 0; i < 1000; i++ {
		slice[i] = i + 1
	}
	
	result := sumSlice(slice, 0)
	fmt.Println(result)
}

File: benchmarks/rust/14_recursive_list_sum.rs
// Recursive List Sum Benchmark - Rust Implementation
// Recursive sum of a vector

fn sum_vec(vec: &Vec<i64>, idx: usize) -> i64 {
    if idx >= vec.len() {
        return 0;
    }
    vec[idx] + sum_vec(vec, idx + 1)
}

fn main() {
    let vec: Vec<i64> = (1..=1000).collect();
    
    let result = sum_vec(&vec, 0);
    println!("{}", result);
}